easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb

```diff
@@ -0,0 +1,52 @@
+easyblock = 'ConfigureMake'
+
+name = 'MBX'
+version = '1.1.0'
+
+homepage = 'https://github.com/paesanilab/MBX'
+description = "MBX is an energy and force calculator for data-driven many-body simulations"
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+source_urls = ['https://github.com/paesanilab/MBX/archive/']
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCE_TAR_GZ}]
+checksums = ['0f55f4950226defb46fd0814ad97f906ce4ffd6403af6817bd98cb3c68996692']
+
+builddependencies = [('Autotools', '20220317')]
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('GSL', '2.7'),
+]
+
+buildininstalldir = True
```

Comment:
I can use `MakeCp`, could you please elucidate me what is wrong with the `buildininstalldir = True`?
I've never use `buildininstalldir` before, I got the easyconfig with this var from @boegel.

Reply:
Using `buildininstalldir` will, as it suggests, unpack and run the configure/build/etc. steps in the install directory.
Depending on the kind of filesystem used to host the install dir, that can be rather slow, especially compared to building on local SSD or in memory.

For builds that don't do configuration that necessitates building in the install dir, it's usually best to avoid doing so

Reply:
Is it ok to stay with easyblock `ConfigureMake`, not use `buildininstalldir = True` and in `postinstallcmds` copy whole MBX-1.1.0 directory to the `installdir`?

Reply:
I'll admit that I got a bit lazy here, and @jfgrimm's remark about building on the filesystem where the installation will be located can be quite slow is very valid.
I'm not sure how relevant it is in this case though, because although the installation takes a long time, it's mostly because the C++ compiler is chewing on individual C++ source files for a very long time, so I/O is a non-issue as far as I could tell.

If we don't enable `buildininstalldir` here, we would need to copy the subdirectory that has the Python bindings after the installation, that could be done via `postinstallcmds` indeed.

I wouldn't copy the whole build directory, only the extra bits that are necessary.

Part of the reason for this approach is that to build the LAMMPS plugin for MBX, you need to copy over some source files from MBX into LAMMPS, see https://github.com/paesanilab/MBX#lammps.

So, in short, I think we can:
* not enable `buildininstalldir`
* copy both `plugins/lammps/USER-MBX` via `postinstallcmds` to the installation directory, as follows:
```python
postinstallcmds = [
    "mkdir -p %(installdir)s/plugins/{lammps,python}",
    "cp -a plugins/lammps/USER-MBX %(installdir)s/plugins/lammps/",
    "cp -a plugins/python/mbx %(installdir)s/plugins/python/",
]
* add `plugins/lammps/USER-MBX` to `sanity_check_paths`